xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+#include "math/color_math.h"
+#include "math/real_math.h"
+
+/* classes */
+
+template <size_t size>
+class c_rasterizer_constant_4f_cache
+{
+    real_vector4d values[size];
+public:
+    bool test_cache(int32 index, real_vector4d* constants, int32 count)
+    {
+        ASSERT(index < size && index + count <= size);
+
+        bool result = false;
+
+        if (count)
+        {
+            bool contents_match = false;
+
+            // Loop through values and constants and xor to determine if values are different
+            int32 i = 0;
+            for (;
+                !(
+                    (uint32)values[index].i ^ (uint32)constants[i].i |
```

Comment:
you convert this value from float to uint, losing the value itself. you need to dereference it as uint, to preserve the actual value...

Reply:
fixed